sktime/forecasting/tests/test_naive.py

```diff
@@ -109,3 +109,48 @@ def test_strategy_mean_seasonal_simple(n_seasons, sp):
     y_pred = f.predict(fh)
 
     np.testing.assert_array_equal(y_pred, expected)
+
+
+@pytest.mark.parametrize("fh", TEST_OOS_FHS)
+@pytest.mark.parametrize("window_length",
+                         [*TEST_WINDOW_LENGTHS, None])
+def test_strategy_drift(fh, window_length):
+    if window_length != 1:
+        f = NaiveForecaster(strategy="drift",
+                            window_length=window_length)
+        f.fit(y_train)
+        y_pred = f.predict(fh)
+
+        if window_length is None:
+            window_length = len(y_train)
+
+        last_window = y_train.iloc[-window_length:].values
+        slope = (last_window[-1] -
+                 last_window[0]) / (len(last_window) - 1)
+        # get well formatted fh values
+        fh = check_fh(fh)
+
+        expected = last_window[-1] + slope * (fh + 1)
+        np.testing.assert_array_equal(y_pred, expected)
+
+
+@pytest.mark.parametrize("fh", TEST_OOS_FHS)
+@pytest.mark.parametrize("window_length",
+                         [*TEST_WINDOW_LENGTHS, None])
+def test_strategy_drift_simple(fh, window_length):
+    # test for flat time series data
+    if window_length != 1:
+        y_train = np.ones((1, 20))
+        f = NaiveForecaster(strategy="drift",
+                            window_length=window_length)
+        f.fit(y_train)
+        y_pred = f.predict(fh)
+
+        if window_length is None:
+            window_length = len(y_train)
+
+        g = NaiveForecaster(strategy="last")
+        g.fit(y_train)
+        expected = g.predict(fh)
```

Comment:
we don't have to use `NaiveForecaster(strategy="last") ` to find out what's expected, no? We can just use `np.ones(len(fh))`, no?

Reply:
adding the other simple case of a line will also be useful: `y_train = pd.Series(np.arange(20))`

Reply:
> adding the other simple case of a line will also be useful: `y_train = pd.Series(np.arange(20))`

It is already implemented in the test_strategy_drift()

Reply:
I see, could you still make it more explicit, in the function name, also we may change the data at the beginning of the script but this function should still work when we change the data, we also don't have to compute the slope, right? We can just use `np.arange(start, end)` where start is the last point in the training series, no?

Reply:
I think start is the successor to last point when we use check_fh